### PR TITLE
For #27101: set dialog to be dismissed when touched outside

### LINKED ISSUE #27101: [Bug]: The wallpaper selection tool cannot be dismissed by tapping outside the window
### Prerequisites: 
The user has just installed the Firefox app for the first time.
The Tab Pickup and Jump Back In CFRs are not currently being displayed.
The user has not seen the wallpaper selection tool before.

### Steps to reproduce

1. Launch the Firefox application.
2. Visit a webpage and then return to Homepage. Repeat this step 3 times.
3. Tap outside the wallpaper selection tool. Observe the behaviour.

### Expected behaviour

The wallpaper selection tool is dismissed by tapping outside the window

### Actual behaviour

The wallpaper selection tool cannot be dismissed by tapping outside the window

### Device name

Xiaomi Pad5 

### Android version

Android 11

### Firefox release type

Firefox Nightly

### Firefox version

Firefox Nightly 107.0a1 09/21

### Device logs

_No response_

### Additional information

https://user-images.githubusercontent.com/108326843/191682106-f73c3455-812f-441a-b4b3-c0e2c3ded510.mp4

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21629)

### PR DESCRIPTION
**Pull request by MatthewTighe**
https://user-images.githubusercontent.com/10427470/191866294-9d406546-00df-4693-9689-d9171e24dafb.mp4

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #27101

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/onboarding/WallpaperOnboardingDialogFragment.kt [hunk 1]
```diff
@@ -80,6 +80,7 @@ class WallpaperOnboardingDialogFragment : BottomSheetDialogFragment() {
         container: ViewGroup?,
         savedInstanceState: Bundle?,
     ): View = ComposeView(requireContext()).apply {
+        this@WallpaperOnboardingDialogFragment.dialog?.setCanceledOnTouchOutside(true)
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/onboarding/WallpaperOnboardingDialogFragment.kt:83]: what if we just do `isCancelable = true` in `onCreate`?

**Comment by MozillaNoah** [comment]
I thought this behavior was intentional? @mavduevskiy

File: app/src/main/java/org/mozilla/fenix/onboarding/WallpaperOnboardingDialogFragment.kt (hunk 1 above)
Comment (MatthewTighe) [reply app/src/main/java/org/mozilla/fenix/onboarding/WallpaperOnboardingDialogFragment.kt:83]: That's actually not working for me, oddly enough

**Comment by MatthewTighe** [comment]
> I thought this behavior was intentional? 

It looks like the AC on the [original JIRA ticket](https://mozilla-hub.atlassian.net/browse/FNXV2-21134) includes:
"If the user clicks on the homepage or clicks the X, then the selection tool will close"

so I would not expect this behavior to be intentional based on that. We can wait til next week to land this and see if we hear back otherwise

**Review by MozillaNoah — APPROVED** [review]

**Pull request merged**